terraform/modules/cluster/eks.tf

```diff
@@ -44,122 +58,77 @@ module "eks_blueprints" {
-      ipv6_cidr_blocks = ["::/0"]
-    }
-
-    # Allows Control Plane Nodes to talk to Worker nodes on Karpenter ports.
-    # This can be extended further to specific port based on the requirement for others Add-on e.g., metrics-server 4443, spark-operator 8080, etc.
-    # Change this according to your security requirements if needed
-    ingress_nodes_karpenter_port = {
-      description                   = "Cluster API to Nodegroup for Karpenter"
-      protocol                      = "tcp"
-      from_port                     = 8443
-      to_port                       = 8443
-      type                          = "ingress"
-      source_cluster_security_group = true
-    }
-
-    ingress_nodes_load_balancer_controller_port = {
-      description                   = "Cluster API to Nodegroup for Load Balancer Controller"
-      protocol                      = "tcp"
-      from_port                     = 9443
-      to_port                       = 9443
-      type                          = "ingress"
-      source_cluster_security_group = true
-    }
-
-    ingress_nodes_metric_server_port = {
-      description                   = "Cluster API to Nodegroup for Metric Server"
-      protocol                      = "tcp"
-      from_port                     = 4443
-      to_port                       = 4443
-      type                          = "ingress"
-      source_cluster_security_group = true
-    }
   }
 
   # Add karpenter.sh/discovery tag so that we can use this as securityGroupSelector in karpenter provisioner
   node_security_group_tags = {
     "karpenter.sh/discovery" = var.environment_name
   }
 
-  managed_node_groups = {
-    mg_5 = {
-      node_group_name = "managed-ondemand"
-      instance_types  = ["m5.large"]
-      subnet_ids      = local.private_subnet_ids
-      min_size        = local.default_mng_min
-      max_size        = local.default_mng_max
-      desired_size    = local.default_mng_size
+  eks_managed_node_groups = {
+    managed-ondemand = {
+      instance_types = ["m5.large"]
 
-      ami_type        = "AL2_x86_64"
-      release_version = var.ami_release_version
+      min_size     = local.default_mng_min
+      max_size     = local.default_mng_max
+      desired_size = local.default_mng_size
 
-      k8s_labels = {
+      labels = {
         workshop-default = "yes"
-        blocker          = sha1(aws_eks_addon.vpc_cni.id)
       }
     }
 
-    system = {
-      node_group_name = "managed-system"
-      instance_types  = ["m5.large"]
-      subnet_ids      = local.primary_private_subnet_id
-      min_size        = 1
-      max_size        = 2
-      desired_size    = 1
+    managed-system = {
+      instance_types = ["m5.large"]
 
-      ami_type        = "AL2_x86_64"
-      release_version = var.ami_release_version
+      min_size     = 1
+      max_size     = 2
+      desired_size = 1
 
-      k8s_taints = [{ key = "systemComponent", value = "true", effect = "NO_SCHEDULE" }]
```

Comment:
I did have one question - this nodegroup, "managed-system" has a taint, and the nodegroup directly below, "managed-ondemand-tainted", does not despite its name containing `taint`. Should the taint be on that nodegroup?